datatypes-cose/src/commonMain/kotlin/at/asitplus/crypto/datatypes/cose/CoseKey.kt

```diff
@@ -1,117 +1,182 @@
-                return null
+        override fun hashCode(): Int {
+            var result = curve?.hashCode() ?: 0
+            result = 31 * result + (x?.contentHashCode() ?: 0)
+            result = 31 * result + (y?.contentHashCode() ?: 0)
+            result = 31 * result + (d?.contentHashCode() ?: 0)
+            return result
+        }
+
+        override fun toCryptoPublicKey(): CryptoPublicKey? {
+            return let {
+                CryptoPublicKey.Ec.fromCoordinates(
+                    curve = curve?.toJwkCurve() ?: return null,
+                    x = x ?: return null,
+                    y = y ?: return null
+                )
             }
-            val xCoordinate = it.sliceArray(1 until 33)
-            val yCoordinate = it.sliceArray(33 until 65)
-            val keyId = MultibaseHelper.calcKeyId(curve.toJwkCurve(), xCoordinate, yCoordinate)
-            return CoseKey(
-                type = type,
-                keyId = keyId.encodeToByteArray(),
-                algorithm = CoseAlgorithm.ES256,
-                curve = curve,
-                x = xCoordinate,
-                y = yCoordinate,
-            )
         }
+    }
 
-        fun fromCoordinates(
-            type: CoseKeyType,
-            curve: CoseEllipticCurve,
-            x: ByteArray,
-            y: ByteArray
-        ): CoseKey? {
-            if (type != CoseKeyType.EC2 || curve != CoseEllipticCurve.P256) {
-                return null
+    // Implements elliptic curve public key parameters in case of y being a bool value
+    data class EcYBoolParams(
+        val curve: CoseEllipticCurve? = null,
+        val x: ByteArray? = null,
+        val y: Boolean? = null,
+        val d: ByteArray? = null
+    ) : CoseKeyParams() {
+
+        override fun equals(other: Any?): Boolean {
+            if (this === other) return true
+            if (other == null || this::class != other::class) return false
+
+            other as EcYBoolParams
+
+            if (curve != other.curve) return false
+            if (x != null) {
+                if (other.x == null) return false
+                if (!x.contentEquals(other.x)) return false
+            } else if (other.x != null) return false
+            if (y != other.y) return false
+            if (d != null) {
+                if (other.d == null) return false
+                if (!d.contentEquals(other.d)) return false
+            } else if (other.d != null) return false
+
+            return true
+        }
+
+        override fun hashCode(): Int {
+            var result = curve?.hashCode() ?: 0
+            result = 31 * result + (x?.contentHashCode() ?: 0)
+            result = 31 * result + (y?.hashCode() ?: 0)
+            result = 31 * result + (d?.contentHashCode() ?: 0)
+            return result
+        }
+
+        override fun toCryptoPublicKey(): CryptoPublicKey? = TODO()
+
+//        TODO conversion to cryptoPublicKey (needs de-/compression of Y coordinate)
```

Comment:
nope, I overlooked that!

Reply:
Oh, mixed it up with the other one (yes the file needs splitting up!) The code works, because we never use that class, as we don't support point-compression. I'll comment it out for now to avoid further confusion